pkg/diff/sql_generator.go

```diff
@@ -1228,6 +1288,176 @@ func (a *attachPartitionSQLVertexGenerator) GetDeleteDependencies(_ schema.Table
 	return nil
 }
 
+type sequenceSQLVertexGenerator struct {
+	deletedTablesByName map[string]schema.Table
+	tableDiffsByName    map[string]tableDiff
+}
+
+func (s *sequenceSQLVertexGenerator) Add(seq schema.Sequence) ([]Statement, error) {
+	sb := strings.Builder{}
+	sb.WriteString(fmt.Sprintf("CREATE SEQUENCE %s\n", seq.GetFQEscapedName()))
+	sb.WriteString(fmt.Sprintf("\tAS %s\n", seq.Type))
+	sb.WriteString(fmt.Sprintf("\tINCREMENT BY %d\n", seq.Increment))
+	sb.WriteString(fmt.Sprintf("\tMINVALUE %d MAXVALUE %d\n", seq.MinValue, seq.MaxValue))
+	cycleModifier := ""
+	if !seq.Cycle {
+		cycleModifier = "NO "
+	}
+	sb.WriteString(fmt.Sprintf("\tSTART WITH %d CACHE %d %sCYCLE\n", seq.StartValue, seq.CacheSize, cycleModifier))
+
+	var hazards []MigrationHazard
+	if seq.Owner == nil {
```

Comment:
Should there be an OWNED BY statement in the create sequence if there is an owner?

Reply:
Ah is this handled by the sequenceOwnershipSQLVertexGenerator?

Reply:
Yep! It's split up because the ownership statement rarely comes directly after the sequence being created. Serials:
```
CREATE SEQUENCE
CREATE TABLE
ASSIGN OWNERSHIP OF SEQUENCE TO TABLE
```